cnd/src/http_api/swap_resource.rs

```diff
@@ -91,21 +89,42 @@ pub fn build_rfc003_siren_entity(
     let id = swap.swap_id;
 
     with_swap_types!(types, {
-        let state: ROLE = state_store
-            .get(&id)?
-            .ok_or_else(|| anyhow!("state store did not contain an entry for {}", id))?;
+        let swap_has_failed = dependencies.swap_error_state.get(&id);
```

Comment:
I have some re-naming ideas for you.  I've a patch, will push to another branch and link you to it in the comments on 'Finish your review'.  You can take a look and see what you think.

Reply:
Naming improvements are very welcome!